extensions/amp-access/0.1/amp-access.js

```diff
@@ -118,6 +118,12 @@ export class AccessService {
     /** @private {?Promise<string>} */
     this.readerIdPromise_ = null;
 
+    /** @private {!Promise} */
+    this.firstAuthorizationPromise_ = new Promise(resolve => {
```

Comment:
It's not the same kind of promise. Full authorization cycle is a bit different from the actual authorization response. What we need in this case is to reliably know whether the publisher has  seen authorization request. Until they have - this promise remains unresolved. In other words - this promise never fails.